protocol/consistencychecks.go

```diff
@@ -7,32 +7,28 @@ package protocol
 
 import (
 	"bytes"
+	"reflect"
 
 	"github.com/coniks-sys/coniks-go/crypto/sign"
 	"github.com/coniks-sys/coniks-go/crypto/vrf"
 	m "github.com/coniks-sys/coniks-go/merkletree"
 )
 
-const (
-	invalidProof int = iota
-	proofOfAbsence
-	proofOfInclusion
-)
-
-// ConsistencyChecks stores the current consistency check
-// state of a CONIKS client. This includes the current epoch
+// ConsistencyChecks stores the latest consistency check
+// state of a CONIKS client. This includes the latest epoch
 // and SignedTreeRoot value, as well as directory's policies
-// (e.g., wherether the TemporaryBinding extension is being used or not)
+// (e.g., whether the TemporaryBinding extension is being used or not)
 // The client should create a new ConsistencyChecks instance only once,
 // when she registers her binding with a ConiksDirectory.
 // This ConsistencyChecks instance then will be used to verify
 // the returned responses from the ConiksDirectory.
 type ConsistencyChecks struct {
+	oldSTR   *m.SignedTreeRoot // keep a copy of SavedSTR before it gets updated
```

Comment:
Is the idea for `oldSTR` to be a workaround for keeping track when the client has transitioned between epochs? It only seems to be relevant in `verifyFulfilledPromise()`, which we still expect to change when we address #116, right?